test/queue_test.rb

```diff
@@ -129,6 +128,30 @@ def enqueue(method, *args)
     QC.default_queue = nil
   end
 
+  def test_multi_threaded_server
+    adapter1 = QC::ConnAdapter.new
+    adapter2 = QC::ConnAdapter.new
+    q1 = q2 = nil
+
+    QC.default_conn_adapter = adapter1
+
+    t1 = Thread.new do
+      QC.default_conn_adapter = adapter1
+      q1 = QC::Queue.new('queue1').conn_adapter
+    end
+
+    t2 = Thread.new do
+      QC.default_conn_adapter = adapter2
+      sleep 0.01
+      q2 = QC::Queue.new('queue2').conn_adapter
+    end
+
+    t1.join
+    t2.join
+
+    refute_equal q1, q2
```

Comment:
as you are assigning the adapters we could compare them directly and make sure `default_conn_adapter` does not create a new instance of `QC::ConnAdapter`:

```ruby
assert_equal adapter1, q1
assert_equal adapter2, q2
```